azurerm/resource_arm_mssql_database_vulnerability_assessment_rule_baseline.go

```diff
@@ -0,0 +1,232 @@
+		return err
+	}
+
+	// Currently REST API always returns "Default" as baseline that's why the Id is created manually here as a workaround
+	resourceId := fmt.Sprintf("/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Sql/servers/%s/databases/%s/vulnerabilityAssessments/Default/rules/%s/baselines/%s",
+		id.SubscriptionID, id.ResourceGroup, id.Path["servers"], id.Path["databases"], id.Path["rules"], *baselineName)
+
+	d.SetId(resourceId)
+
+	return resourceArmMssqlDatabaseVulnerabilityAssessmentRuleBaselineRead(d, meta)
+}
+
+func resourceArmMssqlDatabaseVulnerabilityAssessmentRuleBaselineRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).sql.DatabaseVulnerabilityAssessmentRuleBaselinesClient
+	ctx := meta.(*ArmClient).StopContext
+
+	log.Printf("[INFO] Reading Azure ARM Vulnerability Assessment Rule Baselines.")
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resourceGroupName := id.ResourceGroup
+	serverName := id.Path["servers"]
+	databaseName := id.Path["databases"]
+	ruleId := id.Path["rules"]
+	baselineName := expandBaselineName(id.Path["baselines"])
+
+	result, err := client.Get(ctx, resourceGroupName, serverName, databaseName, ruleId, *baselineName)
+	if err != nil {
+		if utils.ResponseWasNotFound(result.Response) {
+			log.Printf("[WARN] Vulnerability Assessment Rule Baseline %s not found", id)
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("error making Read request to Vulnerability Assessment Rule Baselines: %+v", err)
+	}
+
+	d.Set("resource_group_name", resourceGroupName)
+	d.Set("server_name", serverName)
+	d.Set("database_name", databaseName)
+	d.Set("rule_id", ruleId)
+	d.Set("baseline_name", *baselineName)
+
+	if baselineResults := result.BaselineResults; baselineResults != nil {
+		d.Set("baseline_result", flattenBaselineResult(baselineResults))
+	}
+
+	return nil
+}
+
+func resourceArmMssqlDatabaseVulnerabilityAssessmentRuleBaselineDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).sql.DatabaseVulnerabilityAssessmentRuleBaselinesClient
+	ctx := meta.(*ArmClient).StopContext
+
+	log.Printf("[INFO] Deleting Azure ARM Vulnerability Assessment Rule Baselines.")
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resourceGroupName := id.ResourceGroup
+	serverName := id.Path["servers"]
+	databaseName := id.Path["databases"]
+	ruleId := id.Path["rules"]
+	baselineName := expandBaselineName(id.Path["baselines"])
+
+	result, err := client.Delete(ctx, resourceGroupName, serverName, databaseName, ruleId, *baselineName)
+	if err != nil {
+		if utils.ResponseWasNotFound(result) {
+			log.Printf("[DEBUG] Vulnerability Assessment Rule Baseline %s not found", id)
+			return nil
+		}
+
+		return fmt.Errorf("error deleting Vulnerability Assessment Rule Baselines: %s", err)
+	}
+
+	return nil
+}
+
+func expandBaselineName(baselineName string) *sql.VulnerabilityAssessmentPolicyBaselineName {
```

Comment:
I don't believe this function is necessary, could we just inline it?

Reply:
done